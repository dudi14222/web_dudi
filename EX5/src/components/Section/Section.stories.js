import React from 'react';
import { storiesOf } from '@kadira/storybook';
import {
  withKnobs,
  text
} from '@kadira/storybook-addon-knobs';

import Section from './'


const storiesSection = storiesOf('Section', module);
storiesSection.addDecorator(withKnobs);

storiesSection.addWithInfo('with some props and children', '', () => {
  const story = <Section
  cName={text('class name', 'some-class-name')}
  >
  <p3>children</p3>
  </Section>;
  return story;
});


