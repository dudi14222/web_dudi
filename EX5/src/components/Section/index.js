import React from 'react';
import classNames from 'classnames';
import './section.css';

const Section = ({ children }) => (
    <div className={classNames(['raised-2','section', classNames])}>
        { children }
    </div>
)

export default Section;