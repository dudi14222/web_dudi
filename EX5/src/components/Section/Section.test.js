import Section from '../Section';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";



describe('Section snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Section />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('Section shallow tests', () => {
    it('should render children when passed in', () => {
        const wrapper = shallow((
            <Section>
                <div className="unique" />
            </Section>
        ));    
        expect(wrapper.contains(<div className="unique" />)).toBe(true);
    });
})



