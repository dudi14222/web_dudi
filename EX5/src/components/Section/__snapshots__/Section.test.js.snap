// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section snapshot tests test snapshot 1`] = `
<div
  className="raised-2 section"
/>
`;
